@shared/api/internal.ts

```diff
@@ -183,32 +183,40 @@ export const confirmPassword = async (
   return response;
 };
 
-export const getAccountBalance = async (
+export const getAccountDetails = async (
   publicKey: string,
-): Promise<{
-  balance: string;
-  isFunded: boolean;
-}> => {
-  let response = {
-    balances: [{ asset_code: "", balance: "" }],
-    isFunded: true,
-  };
+): Promise<AccountDetailsInterface> => {
+  const dataProvider = new DataProvider({
+    serverUrl: NETWORK_URL,
+    accountOrKey: publicKey,
+    networkPassphrase: NETWORK_PASSPHRASE,
+  });
+
+  let payments = null;
+  let balances = null;
+  let isFunded = null;
+
+  try {
+    ({ balances } = await dataProvider.fetchAccountDetails());
+    const transactionData = await dataProvider.fetchPayments({
+      limit: TRANSACTIONS_LIMIT,
+    });
+    payments = transactionData?.records || {};
```

Comment:
We could replace `{}` with `null` in this case if there is no need to know if it was "checked" already.